src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/UseEnvironmentProcessIdTests.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.UseEnvironmentProcessId,
+    Microsoft.NetCore.Analyzers.Runtime.UseEnvironmentProcessIdFixer>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Tasks.DoNotCreateTaskCompletionSourceWithWrongArguments,
+    Microsoft.NetCore.Analyzers.Tasks.DoNotCreateTaskCompletionSourceWithWrongArgumentsFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class UseEnvironmentProcessIdTests
+    {
+        [Fact]
+        public async Task NoDiagnostics_NoEnvironmentProcessId_CSharp()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Diagnostics;
+
+class C
+{
+    void M()
+    {
+        int id = Process.GetCurrentProcess().Id; // assumes Environment.ProcessId doesn't exist
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task NoDiagnostics_CSharp()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Diagnostics;
+
+namespace System
+{
+    public static class Environment
+    {
+        public static int ProcessId => 0;
+    }
+}
+
+class C
+{
+    void M()
+    {
+        int id = C.GetCurrentProcess().Id;
+
+        string name = Process.GetCurrentProcess().ProcessName;
+
+        using (var p = Process.GetCurrentProcess())
+            _ = p.Id;
+    }
+
+    private static C GetCurrentProcess() => new C();
+    public int Id => 0;
+}
+");
+        }
+
+        [Fact]
+        public async Task Diagnostics_FixApplies_CSharp()
+        {
+            await VerifyCS.VerifyCodeFixAsync(
+@"
+using System;
+using System.Diagnostics;
+
+namespace System
+{
+    public static class Environment
+    {
+        public static int ProcessId => 0;
+    }
+}
+
+class C
+{
+    int M()
+    {
+        int pid = [|Process.GetCurrentProcess().Id|];
+        Use([|Process.GetCurrentProcess().Id|]);
+        return [|Process.GetCurrentProcess().Id|];
+    }
+
+    void Use(int pid) {}
+}
+",
+@"
+using System;
+using System.Diagnostics;
+
+namespace System
+{
+    public static class Environment
+    {
+        public static int ProcessId => 0;
+    }
+}
+
+class C
+{
+    int M()
+    {
+        int pid = Environment.ProcessId;
+        Use(Environment.ProcessId);
+        return Environment.ProcessId;
+    }
+
+    void Use(int pid) {}
+}
+");
+        }
+    }
```

Comment:
If you want an interesting "with trivia" replacement, perhaps

```C#
int pid = System.
    Diagnostics.
        Process.
            // Static invocation
            GetCurrentProcess().
                Id;
```

I have no idea what the <strike>right</strike> expected replacement is.

Yeah, my interesting trivia isn't all that interesting.  Waterfall with a comment.

Reply:
Thanks.  My gut is any comments in the middle of such an expression should disappear with the expression, as the purpose of the fixer is to replace the whole thing, and if Process is going away and GetCurrentProcess is going away and Id is going away, it's really unlikely any comment would remain relevant.